modules/core/src/main/scala/net/message/Scram.scala

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2018-2020 by Rob Norris
+// This software is licensed under the MIT License (MIT).
+// For more information see LICENSE or https://opensource.org/licenses/MIT
+
+package skunk.net.message
+
+import scodec.bits.ByteVector
+import scodec.codecs.utf8
+
+/**
+  * Implementation of [RFC5802](https://tools.ietf.org/html/rfc5802) as needed by PostgreSQL.
+  * 
+  * That is, only features used by PostgreSQL are implemented -- e.g., channel binding is not supported and
+  * optional message fields that are omitted by PostgreSQL are not supported.
+  */
+private[skunk] object Scram {
+  private implicit class StringOps(val value: String) extends AnyVal {
+    def bytesUtf8: ByteVector = ByteVector.view(value.getBytes(java.nio.charset.StandardCharsets.UTF_8))
+  }
+
+  private def normalize(value: String): String =
+    com.ongres.saslprep.SaslPrep.saslPrep(value, false)
+
+  def clientFirstBare: ByteVector = {
+    val nonce = {
+      val arr = new Array[Byte](32)
+      java.security.SecureRandom.getInstanceStrong().nextBytes(arr)
+      ByteVector.view(arr).toBase64
+    }
+    clientFirstBareWithNonce(nonce)
+  }
```

Comment:
i suppose for this kind of thing purity isn’t the point but this i assume isn’t pure with the secure random call - would we want to wrap in an effect?

Reply:
I don't know - I originally had this in a `F[_]: Sync` but that percolates up the call stack. Adding a `Random` effect seemed overkill. So I simplified to this version.